src/lib/prov/tpm2/tpm2_ecc/tpm2_ecc.cpp

```diff
@@ -0,0 +1,295 @@
+
+      // For creating an asymmetric key this value is not used.
+      .unique = {.ecc = {}},
+   };
+
+   return create_transient_from_template(
+      ctx, sessions, parent.handles().transient_handle(), key_template, sensitive_data);
+}
+
+namespace {
+
+SignatureAlgorithmSelection make_signature_scheme(std::string_view hash_name) {
+   return {
+      .signature_scheme =
+         TPMT_SIG_SCHEME{
+            .scheme = TPM2_ALG_ECDSA,  // Only support ECDSA
+            .details = {.any = {.hashAlg = get_tpm2_hash_type(hash_name)}},
+         },
+      .hash_name = std::string(hash_name),
+      .padding = std::nullopt,
+   };
+}
+
+size_t signature_length_for_key_handle(const SessionBundle& sessions, const Object& object) {
+   const auto curve_id = object._public_info(sessions, TPM2_ALG_ECDSA).pub->publicArea.parameters.eccDetail.curveID;
+
+   const size_t order_bytes = [&]() -> size_t {
+      switch(curve_id) {
+         case TPM2_ECC_NIST_P192:
+            return 24;
+         case TPM2_ECC_NIST_P224:
+            return 28;
+         case TPM2_ECC_NIST_P256:
+            return 32;
+         case TPM2_ECC_NIST_P384:
+            return 48;
+         case TPM2_ECC_NIST_P521:
+            return 66;  // Rounded up to the next full byte
+         default:
+            throw Invalid_Argument(Botan::fmt("Unsupported ECC curve: {}", curve_id));
+      }
+   }();
+
+   return 2 * order_bytes;
+}
+
+class EC_Signature_Operation final : public Signature_Operation {
+   public:
+      EC_Signature_Operation(const Object& object, const SessionBundle& sessions, std::string_view scheme) :
+            Signature_Operation(object, sessions, make_signature_scheme(scheme)) {}
+
+      size_t signature_length() const override { return signature_length_for_key_handle(sessions(), key_handle()); }
+
+      AlgorithmIdentifier algorithm_identifier() const override {
+         // Copied from ECDSA
+         const std::string full_name = "ECDSA/" + hash_function();
+         const OID oid = OID::from_string(full_name);
+         return AlgorithmIdentifier(oid, AlgorithmIdentifier::USE_EMPTY_PARAM);
+      }
+
+   private:
+      std::vector<uint8_t> marshal_signature(const TPMT_SIGNATURE& signature) const override {
+         BOTAN_STATE_CHECK(signature.sigAlg == TPM2_ALG_ECDSA);
+
+         const auto& r = signature.signature.ecdsa.signatureR;
+         const auto& s = signature.signature.ecdsa.signatureS;
+         const auto half_sig_len = signature_length_for_key_handle(sessions(), key_handle()) / 2;
+         BOTAN_ASSERT_NOMSG(r.size == half_sig_len && s.size == half_sig_len);
+
+         return concat<std::vector<uint8_t>>(as_span(r), as_span(s));
+      }
+};
+
+class EC_Verification_Operation final : public Verification_Operation {
+   public:
+      EC_Verification_Operation(const Object& object, const SessionBundle& sessions, std::string_view padding) :
+            Verification_Operation(object, sessions, make_signature_scheme(padding)) {}
```

Comment:
```suggestion
      EC_Verification_Operation(const Object& object, const SessionBundle& sessions, std::string_view hash) :
            Verification_Operation(object, sessions, make_signature_scheme(hash)) {}
```

... for consistency.